Proveedor productos fr

adtiburon-vt opened the pull request:
[FIX] product_states: Añade sudo al buscar compañías
 [FIX] product_stated: Corrige cuando añadir proveedor LX a los productos
 [IMP] product_states: Añade que al crear un producto se añada proveedor LX si no estamos en LX

adtiburon-vt committed:
[IMP] product_states: Añade que al crear un producto se añada proveedor LX si no estamos en LX

adtiburon-vt committed:
[FIX] product_stated: Corrige cuando añadir proveedor LX a los productos

- Ahora se añade para todas las compañías que no sea LX al crear el producto en cualquier compañía

adtiburon-vt committed:
[FIX] product_states: Añade sudo al buscar compañías

omar7r commented on project-addons/product_states/models/product.py:
```diff
@@ -37,3 +37,27 @@ class ProductTemplate(models.Model):
+        product_supplier_create = self.sudo().env['product.supplierinfo'].create
+        for product in self:
+            for company in dependant_companies:
+                if main_company.partner_id not in product.seller_ids.mapped('name'):
```
Acceder a seller_ids sin sudo no creo que funcione, me imagino que tendrá reglas multicompañía y si se desmarca y se vuelve a marcar se duplicaría o daría error de unicidad

adtiburon-vt committed:
[FIX] product_states: Corrige cuando se crean los proveedores por defecto

- Ahora ya no es solo cuando el producto es vendible
- Ahora comprueba mejor si ya se tiene LX como proveedor
